keras_nlp/layers/token_and_position_embedding_test.py

```diff
@@ -0,0 +1,141 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for TokenAndPositionEmbedding"""
+
+import os
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.layers import TokenAndPositionEmbedding
+
+
+def custom_position_embed_init(shape, dtype=None):
+    count = 1
+    for length in shape:
+        count *= length
+    return tf.reshape(tf.range(count, dtype=dtype), shape)
+
+
+def custom_token_embed_init(shape, dtype=None):
+    count = 1
+    for length in shape:
+        count *= length
+    return tf.ones(shape, dtype=dtype)
+
+
+class TokenAndPositionEmbeddingTest(tf.test.TestCase):
+    def test_get_config_and_from_config(self):
+        token_and_position_embed = TokenAndPositionEmbedding(
+            vocabulary_size=5,
+            max_length=10,
+            embedding_dim=32,
+        )
+
+        config = token_and_position_embed.get_config()
+
+        expected_config_subset = {
+            "embeddings_initializer": keras.initializers.serialize(
+                keras.initializers.GlorotUniform()
+            ),
+            "position_embeddings_initializer": keras.initializers.serialize(
+                keras.initializers.GlorotUniform()
+            ),
+            "embeddings_regularizer": None,
+            "mask_zero": False,
+        }
+
+        self.assertEqual(config, {**config, **expected_config_subset})
+
+        restored_token_and_position_embed = (
+            TokenAndPositionEmbedding.from_config(config)
+        )
+
+        self.assertEqual(
+            restored_token_and_position_embed.get_config(),
+            {**config, **expected_config_subset},
+        )
+
+    def test_ragged_tensor_with_3_dimensions(self):
+        vocabulary_size = 5
+        sequence_length = 4
+        feature_size = 2
+        embedding_dim = 3
+        test_layer = TokenAndPositionEmbedding(
+            vocabulary_size=vocabulary_size,
+            max_length=sequence_length,
+            embedding_dim=embedding_dim,
+            embeddings_initializer=custom_token_embed_init,
+            position_embeddings_initializer=custom_position_embed_init,
+        )
+        # Create a 3-dimensional ragged input (the first dimension is implicit).
+        input_tensor = tf.keras.Input(
+            shape=(None, feature_size), dtype=tf.float32, ragged=True
+        )
+        output_tensor = test_layer(input_tensor)
+        model = tf.keras.Model(input_tensor, output_tensor)
+
+        input_data = tf.ragged.constant(
```

Comment:
This test should probably show just a 2d ragged tensor. `(batch_size, ragged_seq_length)`